src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -431,7 +431,8 @@ public void awaitResponses(boolean logBlockingReadRepairAttempt) throws ReadTime
         // return immediately, or begin a read repair
         if (digestResolver.responsesMatch())
         {
-            setResult(digestResolver.getData());
+            PartitionIterator data = digestResolver.getData();
+            setResult(command.filterForReplicaFilteringProtection().apply(data));
```

Comment:
This fixes a large class of problems where we need to filter, but I'm not entirely sure SRP/paging will work...

Reply:
I think we ultimately only want to use a single partition read command if we have a strict `RowFilter`. It's at least an incremental improvement that doesn't require us to worry too much about the short reads potentially caused by non-strict replica filtering coupled w/ strict coordinator filtering.